components/icon/icon.css

```diff
@@ -2,12 +2,15 @@
 
 @layer components {
   .c-icon {
-    display: inline-flex;
-    fill: currentColor;
+    display: inline-block;
```

Comment:
is there a reason why you didn't use tailwind for the styling here? you can apply the tailwind css styles with the keyword @apply. 
e.g. 
@layer components {
```
  .btn-primary {
    @apply py-2 px-4 bg-blue-500 text-white font-semibold rounded-lg shadow-md hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-blue-400 focus:ring-opacity-75;
  }
```

Reply:
I changed a lot of the Icon component in its own PR.